src/classes/VOL_SharedCode.cls

```diff
@@ -318,7 +325,13 @@ global with sharing class VOL_SharedCode {
             } else { // otherwise use the bucket account (which may be null and imply the 1:1 model in NPSP)
                 contact.AccountId = VOL_SharedCode.SettingsBucketAccountId;                             
             }
-            
+
```

Comment:
won't we also need a similar security check above on line 307 for the update case?

Reply:
Good catch!

We can also add the check for lines: 350, 540, and 592.

I'm going re-run PMD. The report is a little noisy because it doesn't recognize our CRUD/FLS checks. But it's a way to flag other dml checks that may have been missed.